### PR TITLE
Throw an error on windows builds with no target specified

### PR DESCRIPTION
**Pull request by Nerzal**
The error message is open for discussion :) 

Fixes: #1748

### CODE REVIEW

**Review by aykevl — APPROVED** [review]
Looks good to me.

File: compileopts/target.go [hunk 1]
```diff
@@ -229,7 +229,13 @@ func LoadTarget(target string) (*TargetSpec, error) {
 	}
 }
 
+// WindowsBuildNotSupportedErr is being thrown, when goos is windows and no target has been specified.
+var WindowsBuildNotSupportedErr = errors.New("Building windows binaries is currently not supported. Try specifying a target")
```
Comment (deadprogram) [inline compileopts/target.go]: How about this error message:

`"Building Windows binaries is currently not supported. Try specifying a different target"`

Comment (Nerzal) [reply compileopts/target.go]: Sounds good, i'll Update it

Comment (deadprogram) [reply compileopts/target.go]: If you can please squash the commit with that new change when you make together with the existing commit, that would be most appreciated.

Comment (Nerzal) [reply compileopts/target.go]: fixed and squashed them

**Comment by Nerzal** [comment]
TestFail seems unrelated

```bash
/go/bin/tinygo test hash/crc64
error: failed to link /tmp/tinygo519719907/main: fork/exec /usr/bin/cc: resource temporarily unavailable
make: *** [Makefile:196: tinygo-test] Error 1
```

**Comment by deadprogram** [comment]
I am re-running it now.

**Comment by deadprogram** [comment]
Passing now. Some spurious error on CircleCI, I suppose.

Now merging, thanks for another cool contribution @Nerzal

**Pull request merged**